data/content/environment_variables.yaml

```diff
@@ -295,6 +295,14 @@ variables:
     A colon separated list of the pipeline's non-private team slugs.
   modifiable: false
   example: "deploy:ops:production"
+- name: BUILDKITE_PLUGIN_CONFIGURATION
+  desc: |
+    JSON string holding the current plugin's configuration (as opposed to all the plugin configurations in the `BUILDKITE_PLUGINS` environment variable).
+  example: "{\"array-key\":[{\"subkey\":[1,2,\"llamas\"]}]}"
+- name: BUILDKITE_PLUGIN_NAME
+  desc: |
+    The "env key-compatible" version of the current plugin name. Used for constructing the `BUILDKITE_PLUGIN_<name>_<param>` variables.
```

Comment:
I find this a bit confusing because I can't see "env key-compatible" naming or `BUILDKITE_PLUGIN_<name>_<param>` described on this page. Looking at [this page](https://github.com/buildkite/docs/blob/main/pages/plugins/writing.md.erb#L49), maybe something like:

> The current plugin's name (as opposed to the names of all plugins). The value matches the GitHub repository slug, not the name defined in `plugin.yml`. You can use the value to access environment variables specific to the plugin using the format: `BUILDKITE_PLUGIN_<PLUGIN_NAME>_<CONFIGURATION_PROPERTY>`.

What do you think?

Reply:
This honestly sounds more confusing to me.

Reply:
My 2 cents here... I also think it's more confusing. Maybe not so much explanation, like for example:

> The current plugin's name. The value matches the GitHub repository slug.

I don't think it is necessary to add more information. Keep it simple like we do with the others variables

Reply:
Yeah, from what I can see on the other page I mentioned, we don't have a good way to talk about these values for plugins. The `name` field in a `plugin.yml` can be a combination of upper and lower case letters, and can contain spaces. It _looks_ like this value is an uppercase version of the name, with spaces replaced with underscores. It's different from the GitHub slug because it's encouraged for that to include the suffix `-buildkite-plugin`. Is that right? If so, perhaps include that in the description. For example:

> The current plugin's name, with all letters in uppercase and any spaces replaced with underscores.

What do you think?

Reply:
Looks good!